src/dmd/doc.d

```diff
@@ -2412,6 +2445,31 @@ extern (C++) void highlightText(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
             }
             break;
 
+        case '\\':
+        {
+            /* Escape Markdown special characters */
+            if (inCode || i+1 >= buf.offset)
+                break;
+            char c1 = buf.data[i+1];
+            if (isPunctuation(c1))
+            {
+                buf.remove(i, 1);
+
+                auto se = sc._module.escapetable.escapeChar(c1);
+                if (!se)
+                    se = c1 == '$' ? "$(DOLLAR)".ptr : c1 == ',' ? "$(COMMA)".ptr : null;
+                if (se)
+                {
+                    const len = strlen(se);
+                    buf.remove(i, 1);
+                    i = buf.insert(i, se, len);
+                    i--; // point to ';'
+                }
+            }
+            leadingBlank = false;
+            break;
```

Comment:
Can this be moved to a separate function?

Reply:
There are several sections like this in `highlightText` that could be moved to separate functions. I generally went with the principle that if it was called in multiple places I made it a separate function and otherwise left it alone. However I'd be happy to pull it out if that's preferred.